src/operators/instance-of.ts

```diff
@@ -0,0 +1,17 @@
+import { Type } from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { filter } from 'rxjs/operators';
+import { IEvent } from '../interfaces';
+
+/**
+ * Filter values depending on their instance type (comparison is made
+ * using `instanceof`).
+ *
+ * @param types List of types implementing `IEvent`.
+ *
+ * @return A stream only emitting the filtered instances.
+ */
+export function instanceOf<TInput extends IEvent, TOutput extends IEvent>(...types: Array<Type<TOutput>>) {
```

Comment:
why not just `Type<TOutput>[]` :D?

Reply:
actually, could we name this operator `ofType` (`instanceOf` sounds great as well, I want to leave consistent naming though :D)

Reply:
> why not just `Type<TOutput>[]` :D?

If you're talking about the use of `Array<>`, this is just a habit due to my (usual) ts-lint rules :grin:, fixin' it!

> actually, could we name this operator ofType (instanceOf sounds great as well, I want to leave consistent naming though :D)

As discussed elsewhere, I agree, let's stick with the Angular wordings :innocent: